src/main/java/com/sourcegraph/cody/agent/CodyAgentClient.java

```diff
@@ -26,14 +33,112 @@ public class CodyAgentClient {
   // onSetConfigFeatures
   @Nullable public Consumer<WebviewPostMessageParams> onReceivedWebviewMessage;
 
-  @Nullable public Editor editor;
+  // Callback for the "editTask/didUpdate" notification from the agent.
+  @Nullable private Consumer<EditTask> onEditTaskDidUpdate;
+
+  // Callback for the "editTask/didDelete" notification from the agent.
+  @Nullable private Consumer<EditTask> onEditTaskDidDelete;
+
+  // Callback for the "textDocument/edit" request from the agent.
+  @Nullable private Consumer<TextDocumentEditParams> onTextDocumentEdit;
+
+  // Callback for the "workspace/executeCommand" request from the agent.
+  @Nullable private Consumer<DisplayCodeLensParams> onDisplayCodeLens;
+
+  // Callback for the "workspace/edit" request from the agent.
+  @Nullable private Consumer<WorkspaceEditParams> onWorkspaceEdit;
+
+  public void setOnEditTaskDidUpdate(@Nullable Consumer<EditTask> callback) {
+    onEditTaskDidUpdate = callback;
```

Comment:
This is fine for now, but longer term is this inviting disaster with overwriting a single callback? Should this throw if the callback is already set, or can the wire-up move to some run once initialization step?

Reply:
I would say let's remove `setOnEditTaskDidUpdate` method and move `onEditTaskDidUpdate` update code into `CodyAgentService::onStartup` as other similar callback.
Current code won't work is some scenarios, e.g. after agent restart after a crash or account change.

Reply:
I agree, but there are 8 of these callbacks (I think my PR only added one), and we should rearchitect all of them in one go. I've added [issue 1136](https://github.com/sourcegraph/jetbrains/issues/1136) to track it.